jcl/src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -62,6 +62,12 @@ public boolean waitForReferenceProcessing() throws InterruptedException {
 	 */
 	static private native boolean waitForReferenceProcessingImpl();
 	
+	/* jdk.lang.ref.disableClearBeforeEnqueue property allow reverting to the old behavior(non clear before enqueue) 
+	 *  defer initializing the immutable variable to avoid bootstrap error 
+	 */
+	static class ClearBeforeEnqueue {
+		static final boolean ENABLED = !Boolean.getBoolean("jdk.lang.ref.disableClearBeforeEnqueue");
```

Comment:
Access of a system property generally needs to be in an AccessController.doPrivileged(). Otherwise if the security manager is enabled and then this is called there will be a SecurityException thrown.